ocaml/xapi/xapi_mgmt_iface.ml

```diff
@@ -143,9 +187,12 @@ let next_server_mode ~mgmt_enabled =
 
 let mgmt_is_enabled () = Server.current_mode () = Any
 
-let run ~__context ~mgmt_enabled =
+let run ~__context ?mgmt_enabled () =
+  let mgmt_enabled = Option.value ~default:(mgmt_is_enabled ()) mgmt_enabled in
   Mutex.execute management_m (fun () ->
```

Comment:
If the rest of the function has to be executed under the lock `@@ fun () ->` would work, too.